### PR TITLE
Implemented Docker Container and Github Actions Workflow

### PR DESCRIPTION
**Pull request by vincentlinzhu**
## Type of pull request:
- [ ] Bug fix
- [X] Feature
- [ ] Other

## Description:
- Made a few changes in the lichess-bot and chess-nanoGPT submodules to implement the work directly below this line.
- Wrote a docker file to install requirements in a container.
- Wrote 2 github actions workflows: one for building the container/pushing it to Dockerhub and another for pulling from Dockerhub/running steps to run train.py/lichess-bot.py in it.

## Related Issues:
This PR relates to https://github.com/ezhang7423/lichess-bot/pull/6

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM mcr.microsoft.com/devcontainers/anaconda:0-3
+
+# Set the working directory in the container
+WORKDIR /chess_transformer_mothership
+
+# Install Python dependencies
+RUN pip install torch numpy transformers datasets tiktoken wandb tqdm
```
Comment (ezhang7423) [inline .devcontainer/Dockerfile:7]: can we create a requirements.txt file in the chess-nanogpt repo and then do the same as line 10,11 as below?

File: .github/workflows/build_docker_container.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
```
Comment (ezhang7423) [inline .github/workflows/build_docker_container.yml:23]: add docs for all secrets needed in the transformer mothership README

File: .github/workflows/run_docker_container.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            echo "No need to clone"
+          else
+            curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash
+            sudo apt-get install git-lfs
+            git lfs install
+            git lfs clone https://huggingface.co/datasets/ezipe/adam-chess-data
+          fi  
+
+  run:
+    timeout-minutes: 6 # 24 hours
```
Comment (ezhang7423) [inline .github/workflows/run_docker_container.yml:52]: change

File: .github/workflows/run_docker_container.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    steps:
+      - name: Use python3 to convert number-jobs to a list
+        id: set-matrix
+        run: |
+          # send telegram message that we are starting a job, send the json config
+          matrix_value=$(python3 -c "print(str(list(range(0, ${{ github.event.inputs.number-jobs || env.number-jobs }} ))).replace(' ', ''))")          
+          echo "job_matrix=$matrix_value" >> $GITHUB_OUTPUT
+
+      - name: Check if folder exists
+        run: test -d "${{ github.workspace }}/adam-chess-data"
```
Comment (ezhang7423) [inline .github/workflows/run_docker_container.yml:35]: is this step necessary?

File: .github/workflows/run_docker_container.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            git lfs clone https://huggingface.co/datasets/ezipe/adam-chess-data
+          fi  
+
+  run:
+    timeout-minutes: 6 # 24 hours
+    runs-on: self-hosted
+    container:
+      image: vincentlinzhu11/chess_transformer_mothership:latest
+      volumes:
+        - ${{ github.workspace }}/adam-chess-data/original:/chess-nanoGPT/data/lichess_hf_dataset  # apply a volume mount
```
Comment (ezhang7423) [inline .github/workflows/run_docker_container.yml:57]: this shouldn't map directly into lichess_hf_dataset - it should go into `/dataset`, and then you should symbolic link to the files inside `/dataset`

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (vincentlinzhu) [reply .devcontainer/Dockerfile:7]: Done

File: .github/workflows/build_docker_container.yml (hunk 2 above)
Comment (vincentlinzhu) [reply .github/workflows/build_docker_container.yml:23]: Done

**Pull request closed without merging**